add_new_dc_test.py

```diff
@@ -0,0 +1,120 @@
+
+        # wait for stress to complete
+        self.verify_stress_thread(cs_thread_pool=read_thread)
+        self.verify_stress_thread(cs_thread_pool=write_thread)
+
+        self.verify_data_can_be_read_from_new_dc(new_node)
+        self.log.info("Test completed.")
+
+    def reconfigure_keyspaces_to_use_network_topology_strategy(self, keyspaces: List[str], replication_factors: dict[str, int]) -> None:
+        node = self.db_cluster.nodes[0]
+        self.log.info("Reconfiguring keyspace Replication Strategy")
+        network_topology_strategy = NetworkTopologyReplicationStrategy(
+            **replication_factors)
+        for keyspace in keyspaces:
+            cql = f"ALTER KEYSPACE {keyspace} WITH replication = {network_topology_strategy}"
+            node.run_cqlsh(cql)
+        self.log.info("Replication Strategies for {} reconfigured".format(keyspaces))
+
+    def prewrite_db_with_data(self) -> None:
+        self.log.info("Prewriting database...")
+        stress_cmd = "cassandra-stress write cl=QUORUM n=209 -schema " \
+                     "'replication(strategy=NetworkTopologyStrategy,eu-westscylla_node_west=3) " \
+                     "compaction(strategy=SizeTieredCompactionStrategy)' -port jmx=6868 -mode cql3 native " \
+                     "-rate threads=8 -pop seq=1..209 -col 'n=FIXED(10) size=FIXED(512)' -log interval=5"
+        pre_thread = self.run_stress_thread(stress_cmd=stress_cmd, stats_aggregate_cmds=False, round_robin=False)
+        self.verify_stress_thread(cs_thread_pool=pre_thread)
+        self.log.info("Database pre write completed")
+
+    def start_stress_during_adding_new_dc(self) -> Tuple[CassandraStressThread, CassandraStressThread]:
+        self.log.info("Running stress during adding new DC")
+        node = self.db_cluster.nodes[0].ip_address
+        stress_cmd_read = f"cassandra-stress read cl=LOCAL_QUORUM duration=20m -port jmx=6868 -mode cql3 native" \
+                          f" -rate threads=8 -pop seq=1..209 -col 'n=FIXED(10) size=FIXED(512)' -log interval=5 " \
+                          f"-node {node}"
+        stress_cmd_write = f"cassandra-stress write cl=LOCAL_QUORUM duration=20m -port jmx=6868 -mode cql3 " \
+                           f"native -rate threads=8 -pop seq=1..209 -col 'n=FIXED(10) size=FIXED(512)' -log interval=5 " \
+                           f"-node {node}"
+        read_thread = self.run_stress_thread(stress_cmd=stress_cmd_read, stats_aggregate_cmds=False, round_robin=False)
+        write_thread = self.run_stress_thread(stress_cmd=stress_cmd_write,
+                                              stats_aggregate_cmds=False, round_robin=False)
+        self.log.info("Stress during adding DC started")
+        return read_thread, write_thread
+
+    def add_node_in_new_dc(self) -> BaseNode:
+        self.log.info("Adding new node")
+        new_node = self.db_cluster.add_nodes(1, dc_idx=1, enable_auto_bootstrap=True)[0]  # add node
```

Comment:
in this case, you are adding a new node, to an existing (but empty) DC...
the idea is to create a new DC, and add a new node to it

Reply:
from cluster perspective, this DC does not exist. Adding DC with 0 nodes shouldn't affect existing DC's nodes in any way. This approach makes everything much easier and more future proof (when sct changes).